src/inmanta/data/__init__.py

```diff
@@ -72,6 +75,9 @@
 # TODO: disconnect
 # TODO: difference between None and not set
 
+# Used as the 'default' parameter value for the Field class, when no default value has been set
+default_unset = object()
```

Comment:
Ah, is it because `None` could be a valid default value? Could you add the motivation to the comment?

Side note (irrelevant but fun (imo)): one of the things I really like about Haskell's type system is that its values actually contain the type information. The `Maybe Int` type has as values `Nothing` or `Just x`. This means you could easily define a `Maybe (Maybe Int)` type as well and have layers of optionals: an absence of a default would be `Nothing` while `Just Nothing` would represent a valid default of `Nothing`. I often miss this sort of thing in Python, where `Optional[Optional[int]]` doesn't make any sense because you can't unambigiously represent those values.

Reply:
Well, this comes from inmanta-lsm and I think @wouterdb is the culprit

Reply:
It is not used in core, so maybe I should just remove it.

Reply:
Isn't it used in core? It isn't to differentiate between no default and a default of `None` then?

Reply:
It is used. The goal is to differentiate between None as default and no default. Why this is used instead of a boolean, you will have to ask @wouterdb

Reply:
My guess would be because you would then need to pass an additional argument.